### DISCUSSION TITLE
실행 컨텍스트 내의 this 바인딩 규칙에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#84] [실행컨텍스트]
- 실행 컨텍스트 내의 this 바인딩 규칙에 대해 설명해 주세요.

**Comment by pipisebastian** [comment]
>실행 컨텍스트 내의 this 바인딩 규칙에 대해 설명해 주세요. 
1. this 는 실행컨텍스트가 생성될때, 즉 함수가 호출되는 시점에 동적으로 정해지는데요
    1. **전역 공간**에서는 전역 객체를 가리킵니다. 브라우저 : `window`, nodeJs : `global`
    2. **함수로써 호출**할때는 전역 객체를 가리킵니다.
    3. **메서드로써 호출**할때는 메서드를 호출한 주체를 가리킵니다. `a.b();`  , `a` ← this            
    4. **생성자함수로써 호출**할때는 새로 만든 인스턴스를 가리킵니다. (new)
    5. +) **콜백함수에서의 this** :  기본적으로 콜백 함수를 함수로써 호출할 경우, 전역객체가 들어가나, call을 통해 명시적으로 this를 바인딩할 수도 있다.

**Comment by publdaze** [comment]
> - 실행 컨텍스트 내의 this 바인딩 규칙에 대해 설명해 주세요.
- 함수 호출 방식에 따라 함수 호출 시점에 동적으로 결정
- 일반 함수 호출 시 - this에 전역 객체 바인딩 (strict mode - undefined)
- 메서드 호출 시 - this에 메서드를 호출한 객체 바인딩
- 생성자 함수 호출 시 - this에 생성자 함수가 생성한 인스턴스 바인딩
- apply, call, bind 메서드에 의한 간접 호출 시 - this에 첫번째 인수로 전달한 객체 바인딩
